lib/srv/db/postgres/users.go

```diff
@@ -532,3 +543,53 @@ var (
 		removePermissionsProcName: removePermissionsProc,
 	}
 )
+
+// withRetry is a helper for auto user teardown that runs a given func a finite
+// number of times until it returns nil error or the given context is done.
+func withRetry(ctx context.Context, log logrus.FieldLogger, f func() error) error {
+	linear, err := retryutils.NewLinear(retryutils.LinearConfig{
+		// arbitrarily copied settings from retry logic in lib/backend/pgbk.
+		First:  0,
+		Step:   100 * time.Millisecond,
+		Max:    750 * time.Millisecond,
```

Comment:
I would give it a little more time (double it?). Unlike our pgbk which is supposed to be fast, I don't trust our connections to Redshift.

Reply:
the max duration is the max time to wait after a failure before retrying, so it shouldn't matter what the connection latency is

Reply:
ya, you are right.